app/main.py

```diff
@@ -1,34 +1,76 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
+
+    def __getitem__(self, item: int) -> int:
+        if item == 0:
+            return self.row
+        elif item == 1:
+            return self.column
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
+        self.is_drowned = is_drowned
+        self.decks = []
+        if start[0] == end[0]:
+            start_point = start[1]
+            end_point = end[1] + 1
+            for column in range(start_point, end_point):
+                self.decks.append(Deck(start[0], column))
+        elif start[1] == end[1]:
+            start_point = start[0]
+            end_point = end[0] + 1
+            for row in range(start_point, end_point):
+                self.decks.append(Deck(row, end[1]))
+        elif start[0] == end[0] and start[1] == end[1]:
+            self.decks.append(Deck(start[0], end[0]))
 
-    def get_deck(self, row, column):
+    def get_deck(self, row: int, column: int) -> Deck:
         # Find the corresponding deck in the list
-        pass
+        for deck in self.decks:
+            if row == deck[0] and column == deck[1]:
+                return deck
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self, row: int, column: int) -> None:
+        deck = self.decks.index(self.get_deck(row, column))
+        self.decks[deck].is_alive = False
+        cheker = False
+        for decks in self.decks:
+            if decks.is_alive is True:
+                cheker = True
+                break
+        if cheker is True:
+            self.is_drowned = False
+        else:
+            self.is_drowned = True
```

Comment:
You could rewrite this code in one line:
```suggestion
        self.is_drowned = not any(deck.is_alive for deck in self.decks)
```